src/precice/impl/DataContext.cpp

```diff
@@ -16,71 +15,98 @@ DataContext::DataContext(mesh::PtrData data, mesh::PtrMesh mesh)
 
 mesh::PtrData DataContext::providedData()
 {
+  PRECICE_TRACE();
   PRECICE_ASSERT(_providedData);
   return _providedData;
 }
 
-mesh::PtrData DataContext::toData()
-{
-  PRECICE_ASSERT(_toData);
-  return _toData;
-}
-
 std::string DataContext::getDataName() const
 {
+  PRECICE_TRACE();
   PRECICE_ASSERT(_providedData);
   return _providedData->getName();
 }
 
 int DataContext::getProvidedDataID() const
 {
+  PRECICE_TRACE();
   PRECICE_ASSERT(_providedData);
   return _providedData->getID();
 }
 
 int DataContext::getFromDataID() const
 {
+  PRECICE_TRACE();
   PRECICE_ASSERT(hasMapping());
   PRECICE_ASSERT(_fromData);
   return _fromData->getID();
 }
 
 void DataContext::resetProvidedData()
 {
+  PRECICE_TRACE();
   _providedData->toZero();
+  // TODO: consistently reset waveform
+  // _providedData->waveform()->toZero();
```

Comment:
The function `SolverInterfaceImpl::resetWrittenData` is calling this function. I tried removing it and all tests still succeed. Since `SolverInterfaceImpl::resetWrittenData` is really ancient (https://github.com/precice/precice/blame/1240c241e6f388b0c9e839c184258f434a9e9d62/src/precice/impl/SolverInterfaceImpl.cpp#L400), but does not add any obvious functionality, I would suggest removing the call. This will also resolve this issue. I will open a corresponding PR.

Reply:
I created the PR #1157.